mutagen/_riff.py

```diff
@@ -0,0 +1,211 @@
+            raise ValueError
+
+        self.__fileobj.seek(self.data_offset)
+        self.__fileobj.write(data)
+
+    def delete(self):
+        """Removes the chunk from the file"""
+
+        delete_bytes(self.__fileobj, self.size, self.offset)
+        if self.parent_chunk is not None:
+            self.parent_chunk._update_size(
+                self.parent_chunk.data_size - self.size)
+
+    def _update_size(self, data_size):
+        """Update the size of the chunk"""
+
+        self.__fileobj.seek(self.offset + 4)
+        self.__fileobj.write(pack('<I', data_size))
+        if self.parent_chunk is not None:
+            size_diff = self.data_size - data_size
+            self.parent_chunk._update_size(
+                self.parent_chunk.data_size - size_diff)
+        self.data_size = data_size
+        self.size = data_size + self.HEADER_SIZE
+
+    def resize(self, new_data_size):
+        """Resize the file and update the chunk sizes"""
+
+        resize_bytes(
+            self.__fileobj, self.data_size, new_data_size, self.data_offset)
+        self._update_size(new_data_size)
+
+
+class RiffFile(object):
+    """Representation of a RIFF file
+
+        Ref: http://www.johnloomis.org/cpe102/asgn/asgn1/riff.html
+      """
+
+    def __init__(self, fileobj):
+        self._fileobj = fileobj
+        self.__subchunks = {}
+
+        # Reset read pointer to beginning of RIFF file
+        fileobj.seek(0)
+
+        # RIFF Files always start with the RIFF chunk
+        self._riffChunk = RiffChunkHeader(fileobj, parent_chunk=None)
+
+        if (self._riffChunk.id != 'RIFF'):
+            raise KeyError("Root chunk should be a RIFF chunk.")
+
+        # Read the RIFF file Type
+        self.fileType = fileobj.read(4).decode('ascii')
+
+        self.__next_offset = fileobj.tell()
+
+        # Load all RIFF subchunks
+        while True:
+            try:
+                chunk = RiffChunkHeader(fileobj, self._riffChunk)
+            except InvalidChunk:
+                break
+            # Normalize ID3v2-tag-chunk to lowercase
+            if chunk.id == 'ID3':
+                chunk.id = 'id3'
+            self.__subchunks[chunk.id] = \
+                chunk
+
+            # Calculate the location of the next chunk,
+            # considering the pad byte
+            self.__next_offset = chunk.offset + chunk.size
+            self.__next_offset += self.__next_offset % 2
+            fileobj.seek(self.__next_offset)
+
+    def __contains__(self, id_):
+        """Check if the IFF file contains a specific chunk"""
+
+        assert_valid_chunk_id(id_)
+
+        return id_ in self.__subchunks
+
+    def __getitem__(self, id_):
+        """Get a chunk from the IFF file"""
+
+        assert_valid_chunk_id(id_)
+
+        try:
+            return self.__subchunks[id_]
+        except KeyError:
+            raise KeyError(
+                "%r has no %r chunk" % (self._fileobj, id_))
+
+    def __delitem__(self, id_):
```

Comment:
I know aiff has the same code, but can you change this to delete_chunk(). It's confusing that `del foo['id']` writes to the file and can raise IOError.